### PR TITLE
feat(cli): prioritize args over config (fix #710)

### LINKED ISSUE #710: Watch mode behaviour can't be override from CLI
### Describe the bug

Watch mode behaviour can't be override by passing flags from CLI

### Reproduction

Create a basic test app from docs

Disable watch mode in config

```ts
/// <reference types="vitest" />
import { defineConfig } from "vite";

export default defineConfig({
  test: {
    watch: false,
  },
});
```

Try running watch mode from CLI

```sh
npx vitest watch
```

or

```sh
npx vitest --watch
```

Screenshots 👇

image

### System Info

```shell
System:
    OS: macOS 12.1
    CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
    Memory: 275.97 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 17.4.0 - /usr/local/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.4.1 - /usr/local/bin/npm
  Browsers:
    Chrome: 98.0.4758.80
    Safari: 15.2
  npmPackages:
    vitest: ^0.2.7 => 0.2.7
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by DerYeger**
Closes #710.

To prioritise CLI args over the config, we have to merge the former last.
However, our usage of CLI default args would result in the test config always being override if an option has a default value set.
To prevent this, I extracted the CLI default args and gave them the lowest priority.

### CODE REVIEW

**Comment by sheremet-va** [comment]
Maybe we should wait for https://github.com/vitest-dev/vitest/pull/702 and do it there?

**Comment by DerYeger** [comment]
> Maybe we should wait for #702 and do it there?

Just had a look at #702. I think it there's no real overlap here.
The issue this PR addresses in particular are the CLI default options, since they have to have a separate priority than the custom CLI options.

Some noteworthy points:
- #702 is missing `allowOnly`
- #702 has a wrong value for `watch`. It shouldn't be `true` unconditionally

I'll create a PR to #702

File: packages/vitest/src/node/cli.ts [hunk 1]
```diff
@@ -16,21 +16,21 @@ cli
   .option('-w, --watch', 'watch mode')
   .option('-t, --testNamePattern <pattern>', 'run test names with the specified pattern')
   .option('--ui', 'enable UI')
-  .option('--open', 'open UI automatically', { default: true })
+  .option('--open', 'open UI automatically')
```
Comment (Demivan) [inline packages/vitest/src/node/cli.ts:19]: This will remove `(default: true)` text from `vitest --help` output. We can add this to option description.

Comment (DerYeger) [reply packages/vitest/src/node/cli.ts:19]: I'll add them

**Comment by DerYeger** [comment]
Will be merged as part of #702

**Pull request closed without merging**